sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts

```diff
@@ -356,15 +356,17 @@ export class LocalTxnDataSerializer implements TxnDataSerializer {
       );
     }
     return {
-      kind: tx,
-      sender: signerAddress,
-      gasData: {
-        payment: gasPayment!,
-        price: originalTx.data.gasPrice!,
-        budget: originalTx.data.gasBudget!,
-        owner: signerAddress,
-      },
-      expiration: { None: null },
+      V1: {
```

Comment:
On the rust side, can you mark this as internally tagged `#[serde(tag = "protocolVesion")]`: https://serde.rs/enum-representations.html#internally-tagged ?

So that this will deserialized to 

``` 
{
 protocolVersion: "V1",
 kind: tx,
 sender: signer,
.....
}

```

Reply:
cc @patrickkuo @Jordan-Mysten

Reply:
I fixed this by translating to and from BCS- and TypeScript- friendly layouts at the serialization boundary.